Sources/AsyncDNSResolver/AsyncDNSResolver.swift

```diff
@@ -126,47 +126,47 @@ public protocol DNSResolver {
     /// - Parameters:
     ///   - name: The name to resolve.
     ///
-    /// - Returns: CNAME for the given name.
-    func queryCNAME(name: String) async throws -> String
+    /// - Returns: CNAME for the given name, `nil` if no record was found.
+    func queryCNAME(name: String) async throws -> String?
 
     /// Lookup SOA record associated with `name`.
     ///
     /// - Parameters:
     ///   - name: The name to resolve.
     ///
-    /// - Returns: ``SOARecord`` for the given name.
-    func querySOA(name: String) async throws -> SOARecord
+    /// - Returns: ``SOARecord`` for the given name, `nil` if no record was found.
+    func querySOA(name: String) async throws -> SOARecord?
 
     /// Lookup PTR record associated with `name`.
     ///
     /// - Parameters:
     ///   - name: The name to resolve.
     ///
-    /// - Returns: ``PTRRecord`` for the given name.
+    /// - Returns: ``PTRRecord`` for the given name, `nil` if no record was found.
```

Comment:
I guess in case of `PTRRecord`, `names` is empty if no records found?

Reply:
Oh yes, good catch, the doc is wrong here.